torchgeo/datasets/splits.py

```diff
@@ -0,0 +1,359 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Dataset splitting utilities."""
+
+from copy import deepcopy
+from itertools import accumulate
+from math import floor
+from typing import List, Optional, Sequence, Tuple, Union
+
+from rtree.index import Index, Property
+from torch import Generator, default_generator, randint, randperm
+
+from ..datasets import GeoDataset
+from .utils import BoundingBox
+
+__all__ = (
+    "random_bbox_assignment",
+    "random_bbox_splitting",
+    "random_grid_cell_assignment",
+    "roi_split",
+    "time_series_split",
+)
+
+
+def _fractions_to_lengths(fractions: Sequence[float], total: int) -> Sequence[int]:
+    """Utility to divide a number into a list of integers according to fractions.
+
+    Args:
+        fractions: list of fractions
+        total: total to be divided
+
+    Returns:
+        List of lengths.
+
+    .. versionadded:: 0.5
+    """
+    lengths = [floor(frac * total) for frac in fractions]
+    remainder = int(total - sum(lengths))
+    # add 1 to all the lengths in round-robin fashion until the remainder is 0
+    for i in range(remainder):
+        idx_to_add_at = i % len(lengths)
+        lengths[idx_to_add_at] += 1
+    return lengths
+
+
+def random_bbox_assignment(
+    dataset: GeoDataset,
+    lengths: Sequence[Union[int, float]],
```

Comment:
In typing, int is a subset of float, so you don't need to union:
```suggestion
    lengths: Sequence[float],
```